pkg/operations/switchover.go

```diff
@@ -198,69 +192,51 @@ func handleSwitchoverProgress(reqCtx intctrlutil.RequestCtx, cli client.Client,
 			Status:    opsv1alpha1.ProcessingProgressStatus,
-			Message:   fmt.Sprintf("waiting for component %s pod role label consistency after switchover", switchover.ComponentName),
+			Message:   fmt.Sprintf("do switchover for component %s and check role label", switchover.ComponentName),
 		}
 		compSpec := opsRes.Cluster.Spec.GetComponentByName(switchover.ComponentName)
-		synthesizedComp, errBuild := buildSynthesizedComp(reqCtx.Ctx, cli, opsRes, compSpec)
-		if errBuild != nil {
-			checkRoleLabelProcessDetail.Message = fmt.Sprintf("handleSwitchoverProgress build synthesizedComponent %s failed", switchover.ComponentName)
-			checkRoleLabelProcessDetail.Status = opsv1alpha1.FailedProgressStatus
-			setComponentSwitchoverProgressDetails(reqCtx.Recorder, opsRequest, appsv1.UpdatingClusterCompPhase, checkRoleLabelProcessDetail, switchover.ComponentName)
-			continue
+		synthesizedComp, err := buildSynthesizedComp(reqCtx.Ctx, cli, opsRes, compSpec)
+		if err != nil {
+			failedCount += 1
+			doNCheckSwitchoverProcessDetail.Message = fmt.Sprintf("component %s do switchover build synthesizedComponent failed", switchover.ComponentName)
+			doNCheckSwitchoverProcessDetail.Status = opsv1alpha1.FailedProgressStatus
+			setComponentSwitchoverProgressDetails(reqCtx.Recorder, opsRequest, appsv1.UpdatingClusterCompPhase, doNCheckSwitchoverProcessDetail, switchover.ComponentName)
+			break
 		}
-		consistency, err = checkPodRoleLabelConsistency(reqCtx.Ctx, cli, *synthesizedComp, &switchover, switchoverCondition)
+		compDef, err := component.GetCompDefByName(reqCtx.Ctx, cli, synthesizedComp.CompDefName)
 		if err != nil {
-			checkRoleLabelProcessDetail.Message = fmt.Sprintf("waiting for component %s pod role label consistency after switchover", switchover.ComponentName)
-			setComponentSwitchoverProgressDetails(reqCtx.Recorder, opsRequest, appsv1.UpdatingClusterCompPhase, checkRoleLabelProcessDetail, switchover.ComponentName)
-			continue
+			failedCount += 1
+			doNCheckSwitchoverProcessDetail.Message = fmt.Sprintf("component %s do switchover get component definition failed", switchover.ComponentName)
+			doNCheckSwitchoverProcessDetail.Status = opsv1alpha1.FailedProgressStatus
+			setComponentSwitchoverProgressDetails(reqCtx.Recorder, opsRequest, appsv1.UpdatingClusterCompPhase, doNCheckSwitchoverProcessDetail, switchover.ComponentName)
+			break
+		}
+		synthesizedComp.TemplateVars, _, err = component.ResolveTemplateNEnvVars(reqCtx.Ctx, cli, synthesizedComp, compDef.Spec.Vars)
+		if err != nil {
+			failedCount += 1
+			doNCheckSwitchoverProcessDetail.Message = fmt.Sprintf("component %s do switchover build synthesizedComponent template vars failed", switchover.ComponentName)
+			doNCheckSwitchoverProcessDetail.Status = opsv1alpha1.FailedProgressStatus
+			setComponentSwitchoverProgressDetails(reqCtx.Recorder, opsRequest, appsv1.UpdatingClusterCompPhase, doNCheckSwitchoverProcessDetail, switchover.ComponentName)
```

Comment:
why not replace to a function for error handle

Reply:
> why not replace to a function for error handle

done